terraform/000-core/main.tf

```diff
@@ -56,3 +56,81 @@ resource "aws_iam_user_policy" "cd_ecs" {
 }
 EOF
 }
+
+/*
+ * Create CloudTrail resources
+ */
+resource "aws_s3_bucket" "cloudtrail" {
+  bucket        = "${var.app_name}-${var.app_env}-cloudtrail"
+  force_destroy = true
+
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AWSCloudTrailAclCheck",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:GetBucketAcl",
+            "Resource": "arn:aws:s3:::${var.app_name}-${var.app_env}-cloudtrail"
+        },
+        {
+            "Sid": "AWSCloudTrailWrite",
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "s3:PutObject",
+            "Resource": "arn:aws:s3:::${var.app_name}-${var.app_env}-cloudtrail/*",
+            "Condition": {
+                "StringEquals": {
+                    "s3:x-amz-acl": "bucket-owner-full-control"
+                }
+            }
+        }
+    ]
+}
+POLICY
+}
+
+resource "aws_iam_user" "cloudtrail-s3" {
+  name = "cloudtrail-s3-${var.app_name}-${var.app_env}"
+}
+
+resource "aws_iam_access_key" "cloudtrail-s3" {
+  user = "${aws_iam_user.cloudtrail-s3.name}"
+}
+
+resource "aws_iam_user_policy" "cloudtrail-s3" {
+  name = "ECS-ECR"
```

Comment:
Does this policy name need to be dynamic or IdP-specific in some way?

Reply:
I thought so for a moment, but realized it is a policy attached to the user, not a general/generic policy so it does not need to be unique across the account